service/topology_coordinator.cc

```diff
@@ -2781,6 +2787,13 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                 const auto node_rs_state = node.rs->state;
                 const auto node_id = node.id;
 
+                const bool is_removenode = node_rs_state == node_state::removing;
```

Comment:
A bit strange to have `is_removenode` when we already have `node_rs_state`. `is_removenode` is used in 4 places, so maybe it's fine.